src/go/rpk/pkg/cli/cmd/debug/info.go

```diff
@@ -276,7 +280,21 @@ func getKafkaInfo(
 		conf.Redpanda.KafkaApi[0].Address,
 		conf.Redpanda.KafkaApi[0].Port,
 	)
-	client, err := kafka.InitClientWithConf(&conf.Rpk.TLS, &conf.Rpk.SCRAM, addr)
+	var tlsConfig *tls.Config
+	if conf.Rpk.KafkaApi.TLS != nil {
+		t := conf.Rpk.KafkaApi.TLS
+		tlsConfig, err = vtls.BuildTLSConfig(fs, true, t.CertFile, t.KeyFile, t.TruststoreFile)
+		if err != nil {
+			out <- [][]string{}
+			kafkaInfoCh <- kInfo
+			return errors.Wrap(err, "Error loading TLS configuration")
+		}
+	}
```

Comment:
Does this need `else if conf.Rpk.TLS != nil` to check for the old (backcompat) behavior?

Reply:
Yes, you're right. I will push it after I finish testing everything.